### ISSUE TITLE
why the code execution of cosin similarity in the optimization of manger and worker is 1 - cos(dis_feat, goal)?

### ISSUE THREAD

**Issue by qingchenwuhou** [CR-Gjx/LeakGAN#9]
Hi, jiaxian: 
  Thanks for your great work.
   I have a question that why the code execution of cosin similarity in the optimization of manger and worker is 1 - cos(dis_feat, goal) rather than cos(dis_feat, goal).
  For example:
280         self.Worker_Reward = 1 - tf.losses.cosine_distance(tf.nn.l2_normalize(self.all_sub_features,2), tf.nn.l2_normalize(self.all_sub_goals,2), 2)

**Comment by qingchenwuhou** [comment]
280 self.Worker_Reward = 1 - tf.losses.cosine_distance(tf.nn.l2_normalize(self.all_sub_features,2), tf.nn.l2_normalize(self.all_sub_goals,2), 2)
the up code is the objection function of manager, and the corresponding gradient is equation（7）
Solved.
